### PR TITLE
[CNSMR-3229] Add development process

### PR DESCRIPTION
**Pull request by danspndl**
Added documentation describing our development process from picking up a ticket, to releasing it

### CODE REVIEW

File: Cookbook/Technical-Documents/Development-Process.md
Comment (anilputtabuddhi) [inline Cookbook/Technical-Documents/Development-Process.md]: Would a link to some of our architecture patterns like Unidirectional flow, Reactive Swift, Bento, ReactiveFeedback etc under this section also be useful?  I think "Get Started with development"  could be split into "Git setup and feature flagging" and then "Development"(which includes links to best practices, architecture documentation, meeting A/C etc).

File: Cookbook/Technical-Documents/Development-Process.md
Comment (AnnKatF) [inline Cookbook/Technical-Documents/Development-Process.md]:
```suggestion
## 1. Get a Jira ticket 🎫
```

File: Cookbook/Technical-Documents/Development-Process.md
Comment (AnnKatF) [inline Cookbook/Technical-Documents/Development-Process.md]:
```suggestion
- The ticket must be assigned to the engineer who is going to work on it. The engineer should make sure it has all details and designs attached (if applicable).
```

File: Cookbook/Technical-Documents/Development-Process.md
Comment (AnnKatF) [inline Cookbook/Technical-Documents/Development-Process.md]:
```suggestion
- The ticket will be moved to `In progress` once development is ready to start (this might be automatically done by the bot when branch published, if configured).
```

File: Cookbook/Technical-Documents/Development-Process.md
Comment (AnnKatF) [inline Cookbook/Technical-Documents/Development-Process.md]:
```suggestion
- Create a branch where this work will live. Please follow this format: `<author's_name>/<ticket-id>-<small-description (optional)>` (E.g. `john/TCKT-123-fix-bug`)
```

File: Cookbook/Technical-Documents/Development-Process.md
Comment (AnnKatF) [inline Cookbook/Technical-Documents/Development-Process.md]: Can we add a link to the `GalleryApp`?

File: Cookbook/Technical-Documents/Development-Process.md
Comment (AnnKatF) [inline Cookbook/Technical-Documents/Development-Process.md]:
```suggestion
## 2. Get started with development 🛠
```

File: Cookbook/Technical-Documents/Development-Process.md
Comment (AnnKatF) [inline Cookbook/Technical-Documents/Development-Process.md]:
```suggestion
## 3. Testing 🧪 <!-- TODO: [CNSMR-3195] link to testing article -->
```

File: Cookbook/Technical-Documents/Development-Process.md
Comment (AnnKatF) [inline Cookbook/Technical-Documents/Development-Process.md]:
```suggestion
- Our features should include the following type of tests:
```

File: Cookbook/Technical-Documents/Development-Process.md
Comment (AnnKatF) [inline Cookbook/Technical-Documents/Development-Process.md]:
```suggestion
  - Snapshot test should be added to new components and screens
```

File: Cookbook/Technical-Documents/Development-Process.md
Comment (AnnKatF) [inline Cookbook/Technical-Documents/Development-Process.md]:
```suggestion
  - Logic, ViewModels, SDK changes should be unit tested
```

File: Cookbook/Technical-Documents/Development-Process.md
Comment (AnnKatF) [inline Cookbook/Technical-Documents/Development-Process.md]:
```suggestion
  - Automation tests should be updated or added.
```

File: Cookbook/Technical-Documents/Development-Process.md
Comment (AnnKatF) [inline Cookbook/Technical-Documents/Development-Process.md]:
```suggestion
- Please test `GalleryApp` and `Babylon` by building and running before merging to avoid checks failing.
```

File: Cookbook/Technical-Documents/Development-Process.md
Comment (AnnKatF) [inline Cookbook/Technical-Documents/Development-Process.md]:
```suggestion
## 4. Creating the PR 📝
```

File: Cookbook/Technical-Documents/Development-Process.md
Comment (AnnKatF) [inline Cookbook/Technical-Documents/Development-Process.md]:
```suggestion
## 5. Merging PR 🚦
```

File: Cookbook/Technical-Documents/Development-Process.md
Comment (AnnKatF) [inline Cookbook/Technical-Documents/Development-Process.md]:
```suggestion
- Once it reaches the front of the queue, it's updated with the target branch and CircleCI runs all necessary tests to make sure nothing's failing. _Note: there's no need to trigger checks manually, if there are other PRs in the queue._ <!-- TODO: [CNSMR-3231] link to Wall-E article -->
```

File: Cookbook/Technical-Documents/Development-Process.md
Comment (AnnKatF) [inline Cookbook/Technical-Documents/Development-Process.md]:
```suggestion
- If the required checks fail you will have to analyse what is the reason for the failure, fix it and add the `Merge` label again.
```

File: Cookbook/Technical-Documents/Development-Process.md
Comment (AnnKatF) [inline Cookbook/Technical-Documents/Development-Process.md]:
```suggestion
## 6. QA 🧑‍💻
```

File: Cookbook/Technical-Documents/Development-Process.md
Comment (AnnKatF) [inline Cookbook/Technical-Documents/Development-Process.md]:
```suggestion
## 7. Release 🚢
```

File: Cookbook/Technical-Documents/Development-Process.md
Comment (AnnKatF) [inline Cookbook/Technical-Documents/Development-Process.md]:
```suggestion
- When releasing a feature that was under a local feature flag, make sure it is removed **before** the release branch is cut
```

**Review by AnnKatF — COMMENTED** [review]
Thank you for writing this document.
It has a good structure.
I've left some comments.

File: Cookbook/Technical-Documents/Development-Process.md
Comment (sdaniel55) [reply Cookbook/Technical-Documents/Development-Process.md]: Wasn't sure how detailed this article should be. I think if someone's looking for code guides, they can find it in the Table of Contents, otherwise when we create or edit articles, it'd need to be updated in both places (and there's a good chance it won't happen). I'd rather keep this simple and generic

File: Cookbook/Technical-Documents/Development-Process.md
Comment (sdaniel55) [reply Cookbook/Technical-Documents/Development-Process.md]: In App Center? Or do we have an article about it?

File: Cookbook/Technical-Documents/Development-Process.md
Comment (AnnKatF) [reply Cookbook/Technical-Documents/Development-Process.md]: I agree with you Dan. We can mention:
> the implementation should be based in our current [Architecture](https://github.com/babylonhealth/babylon-ios/blob/develop/Documentation/README.md#architecture-)

File: Cookbook/Technical-Documents/Development-Process.md
Comment (sdaniel55) [reply Cookbook/Technical-Documents/Development-Process.md]: Checking the "required checks" part, there was a recent change in Wall-E that might have changed this

File: Cookbook/Technical-Documents/Development-Process.md
Comment (AnnKatF) [reply Cookbook/Technical-Documents/Development-Process.md]: App Center as there is not article for it ¯\_(ツ)_/¯

Comment (sdaniel55) [reply Cookbook/Technical-Documents/Development-Process.md]: Fixed in 15b472e

File: Cookbook/Technical-Documents/Development-Process.md
Comment (sdaniel55) [reply Cookbook/Technical-Documents/Development-Process.md]: Just going to wait until #301 is merged so I can link to the section :)

File: Cookbook/Technical-Documents/Development-Process.md
Comment (sdaniel55) [reply Cookbook/Technical-Documents/Development-Process.md]: All checks are needed, not just required ones
```suggestion
- If the checks fail, you will have to analyse what is the reason for the failure, fix it, and add the `Merge` label again
```

File: Cookbook/Technical-Documents/Development-Process.md
Comment (AnnKatF) [reply Cookbook/Technical-Documents/Development-Process.md]: yeah makes sense :)

**Review by AnnKatF — APPROVED** [review]
Looks good 👍 
🚀

File: Cookbook/Technical-Documents/Development-Process.md
Comment (chitrakotwani) [inline Cookbook/Technical-Documents/Development-Process.md]:
```suggestion
- The ticket will be moved to `In progress` once development is ready to start (this might be automatically done by the bot when branch is published, if configured)
```

**Review by chitrakotwani — APPROVED** [review]

**Pull request merged**
